model/src/main/java/org/projectnessie/model/IcebergTable.java

```diff
@@ -60,22 +71,39 @@ public abstract class IcebergTable extends Content {
   @NotBlank
   public abstract String getMetadataLocation();
 
+  /** Corresponds to Iceberg's {@code currentSnapshotId}. */
+  @Min(DEFAULT_SNAPSHOT_ID)
   public abstract long getSnapshotId();
 
+  /** Corresponds to Iceberg's {@code currentSchemaId}. */
+  @Min(INITIAL_SCHEMA_ID)
   public abstract int getSchemaId();
 
+  /** Corresponds to Iceberg's {@code defaultSpecId}. */
+  @Min(INITIAL_SPEC_ID)
   public abstract int getSpecId();
 
+  /** Corresponds to Iceberg's {@code defaultSortOrderId}. */
+  @PositiveOrZero
   public abstract int getSortOrderId();
 
   @Override
   public Type getType() {
     return Type.ICEBERG_TABLE;
   }
 
+  @Schema(type = SchemaType.OBJECT)
+  @Nullable
+  @JsonInclude(Include.NON_NULL)
+  public abstract JsonNode getMetadata();
```

Comment:
nit: maybe add a separate Iceberg version to set a context for interpreting this free-form metadata?

Reply:
Moved the `JsonNode` into a new `GenericMetadata` composite, which has a "variant".